contribs/gmf/src/directives/print.js

```diff
@@ -665,7 +681,34 @@ gmf.PrintController.prototype.print = function(format) {
   goog.asserts.assertNumber(this.fields.dpi);
   goog.asserts.assertString(this.fields.layout);
 
-  var spec = this.ngeoPrint_.createSpec(this.map, scale, this.fields.dpi,
+  // convert the WMTS layers to WMS
+  var map = new ol.Map({});
+  map.setView(this.map.getView());
+  var ol_layers = this.ngeoLayerHelper_.getFlatLayers(this.map.getLayerGroup());
+  var new_ol_layers = [];
+  for (var i = 0, ii = ol_layers.length; i < ii; i++) {
+    var layer = ol_layers[i];
+    var metadata = layer.get('metadata');
+    if (metadata) {
+      var server_name = metadata.ogcServer;
+      var layer_names = metadata.layers || metadata.printLayers;
+      if (server_name && layer_names) {
+        var server = this.ogcServers_[server_name];
+        goog.asserts.assert(server);
```

Comment:
In this case we should log an error with production code, the server name isn't verified before ...